HairSuite.Domain/Reservation.cs

```diff
@@ -5,28 +5,36 @@ public class Reservation : Aggregate
     public Guid Id
     {
         get => ReservationId.Value;
-        set {}
+        set { }
     }
+
     public ReservationId ReservationId { get; set; }
     public HairdresserId UserId { get; set; }
     public ReservationDate Date { get; set; }
     public ReservationStatus Status { get; set; }
 
-    public Reservation() { } // For serialization
-
-    private Reservation(Guid id, Guid hairdresserId, DateTime date)
+    // For serialization
+    public Reservation()
     {
-        HandleEvent(new Events.ReservationRequested(id, hairdresserId, date), Apply);
     }
 
+    private Reservation(Guid id, Guid hairdresserId, DateTime date) =>
+        HandleEvent(new Events.ReservationRequested(id, hairdresserId, date), Apply);
+
     public static Reservation MakeTentative(Guid id, Guid hairdresserId, DateTime date) => new(id, hairdresserId, date);
 
-    public void Confirm(Guid id)
+    public void Confirm(Guid id, Func<ReservationId, ReservationDate, bool> isDateReserved)
```

Comment:
Would it be better to use `Func<Reservation, bool>`? This would be less prone to changes even if we decide to change the underlying implementation and need another property. An example of this would be if we want to add a reservation length or an estimated finish time.